### ISSUE TITLE
Incorrect gradient of lax.digamma for negative half-integer inputs

### ISSUE THREAD

**Issue by ALinrunrun** [jax-ml/jax#37635]
## Description
`jax.lax.digamma` appears to produce an incorrect reverse-mode gradient for negative half-integer inputs.

The derivative of the digamma function is the trigamma function, so `jax.jacrev(lambda x: jax.lax.digamma(x))` should agree with `scipy.special.polygamma(1, x)` for finite non-pole inputs.

The tested values below are negative half-integers, not poles of the digamma function. SciPy returns finite positive trigamma values, but JAX returns small negative gradients.

## Minimal reproducible example:

```
import jax
import jax.numpy as jnp
from scipy.special import polygamma

fn = lambda x: jax.lax.digamma(x)

for v in [-16.5, -19.5, -22.5]:
    a = jnp.array([v])
    grad = float(jax.jacrev(fn)(a)[0, 0])
    expected = float(polygamma(1, v))
    print(f"digamma'({v})  jax={grad:+.4e}  scipy={expected:+.4f}")
```

Actual result:

```
An NVIDIA GPU may be present on this machine, but a CUDA-enabled jaxlib is not installed. Falling back to cpu.
digamma'(-16.5)  jax=-5.8807e-02  scipy=+9.8108
digamma'(-19.5)  jax=-4.9990e-02  scipy=+9.8196
digamma'(-22.5)  jax=-4.3471e-02  scipy=+9.8261
```

Expected result:

```
The gradient of jax.lax.digamma should match scipy.special.polygamma(1, x).

Expected values are approximately:
digamma'(-16.5) ~= +9.8108
digamma'(-19.5) ~= +9.8196
digamma'(-22.5) ~= +9.8261
```

### System info (python version, jaxlib version, accelerator, etc.)

```
WARNING:2026-05-07 00:33:38,915:jax._src.xla_bridge:794: An NVIDIA GPU may be present on this machine, but a CUDA-enabled jaxlib is not installed. Falling back to cpu.
jax:    0.6.2
jaxlib: 0.6.2
numpy:  2.2.6
python: 3.10.20 (main, Mar 11 2026, 17:46:40) [GCC 14.3.0]
device info: cpu-1, 1 local devices
process_count: 1
platform: uname_result(system='Linux', node='MSI', release='5.15.167.4-microsoft-standard-WSL2', version='#1 SMP Tue Nov 5 00:21:55 UTC 2024', machine='x86_64')

nvidia-smi:
NVIDIA-SMI 545.23.06
Driver Version: 545.92
CUDA Version: 12.3
GPU: NVIDIA GeForce RTX 3080, 8192 MiB

Note: Reproduced on CPU fallback because CUDA-enabled jaxlib is not installed.
```

**Comment by AnirudhSKrishnan** [comment]
I'm seeing incorrect results from `jax.lax.polygamma(1, x)` for sufficiently negative non-integer inputs, which also affects gradients of `jax.lax.digamma`.

```python
import jax
import jax.numpy as jnp
from scipy.special import polygamma

for v in [-3.3, -7.5, -8.5, -9.5, -16.5]:
    got = float(jax.lax.polygamma(jnp.array(1.0), jnp.array(v)))
    expected = float(polygamma(1, v))
    print(f"x={v:>6}  jax={got:>16.5f}  scipy={expected:>10.5f}")
```

| x | JAX `polygamma(1, x)` | SciPy |
|---:|---:|---:|
| -3.3 | 14.81774 | 14.81774 |
| -7.5 | 9.74476 | 9.74477 |
| -8.5 | 6.79496 | 9.75861 |
| -9.5 | -2854289145856.0 | 9.76969 |
| -16.5 | -0.05881 | 9.81080 |

This does not seem specific to half-integer inputs: non-half-integers such as `-10.7` and `-16.1` also give incorrect results. The results begin to degrade once the input drops below approximately `-8`, and break catastrophically near inputs such as `-9.5` and `-10.5`.

While looking into this, I noticed that these operations lower to `chlo.zeta`, and XLA's `Zeta` implementation computes a fixed number of initial terms followed by an Euler–Maclaurin correction evaluated at `q + 10`:

For the trigamma case, this evaluates `zeta(2, q)`. My understanding is that once `q` is sufficiently negative, the correction is evaluated at a non-positive or very small expansion point. 

I may be misunderstanding either the intended domain of `chlo.zeta` or where this case is expected to be handled, so I wanted to ask before attempting a fix:

- Is this something that should be addressed in JAX, for example by handling negative non-integer inputs through the trigamma reflection formula before lowering?
- Or would the preferred fix be upstream in `chlo.zeta`, since `lax.zeta(2, q)` itself is affected?

For reference, using the reflection identity

```text
ψ₁(x) = π² / sin²(πx) − ψ₁(1 − x)
```

matches SciPy for the failing trigamma cases I tested.

**Comment by bej9** [comment]
This looks like the same issue as #11481
